### PR TITLE
Clean up the endpoint and headers usage

### PR DESCRIPTION
**Pull request by jimirocks**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java/src/main/java/com/gooddata/sdk/service/GoodDataSettings.java [hunk 1]
```diff
@@ -197,7 +216,7 @@ public void setPollSleepSeconds(final int pollSleep) {
      * @return user agent string
      */
     public String getUserAgent() {
-        return userAgent;
+        return StringUtils.isNotBlank(userAgent) ? String.format("%s %s", userAgent, getDefaultUserAgent()) : getDefaultUserAgent();
```
Comment (peter-plochan) [inline gooddata-java/src/main/java/com/gooddata/sdk/service/GoodDataSettings.java]: I'm not sure about use cases, but could you consider to add another method which returns the `userAgent` property without default agent? And maybe add more javadoc what returns this particular method.

File: gooddata-java/src/test/groovy/com/gooddata/sdk/service/export/ExportServiceTest.groovy [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    def "should not create"() {
+        when:
+        new ExportService(null, new GoodDataSettings())
+
+        then:
+        thrown(IllegalArgumentException)
+    }
+
+    def "should fail exportPdf on invalid arguments"() {
```
Comment (peter-plochan) [inline gooddata-java/src/test/groovy/com/gooddata/sdk/service/export/ExportServiceTest.groovy]: missing `@Unroll`

File: gooddata-java/src/test/groovy/com/gooddata/sdk/service/export/ExportServiceTest.groovy [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+        where:
+        endpoint               | dashboardArg | tabArg | output                      | msgRegex
+        null                   | dashboard    | tab    | new ByteArrayOutputStream() | /.*endpoint.*/
+        new GoodDataEndpoint() | null         | tab    | new ByteArrayOutputStream() | /.*dashboard.*/
+        new GoodDataEndpoint() | dashboard    | null   | new ByteArrayOutputStream() | /.*tab.*/
+        new GoodDataEndpoint() | dashboard    | tab    | null                        | /.*output.*/
+    }
+
+    def "should fail exportCsv report on invalid arguments"() {
```
Comment (peter-plochan) [inline gooddata-java/src/test/groovy/com/gooddata/sdk/service/export/ExportServiceTest.groovy]: missing `@Unroll`

File: gooddata-java/src/test/groovy/com/gooddata/sdk/service/export/ExportServiceTest.groovy [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        def exc = thrown(IllegalArgumentException)
+        exc.message =~ msgRegex
+
+        where:
+        toExport | output                      | msgRegex
+        null     | new ByteArrayOutputStream() | /.*report.*/
+        report   | null                        | /.*output.*/
+    }
+
+    def "should fail exportCsv reportDefinition on invalid arguments"() {
```
Comment (peter-plochan) [inline gooddata-java/src/test/groovy/com/gooddata/sdk/service/export/ExportServiceTest.groovy]: missing `@Unroll`

File: gooddata-java/src/test/groovy/com/gooddata/sdk/service/export/ExportServiceTest.groovy [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import com.gooddata.sdk.service.GoodDataEndpoint
+import com.gooddata.sdk.service.GoodDataSettings
+import org.springframework.web.client.RestTemplate
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Subject
+
+import static com.gooddata.util.ResourceUtils.readObjectFromResource
+
+class ExportServiceTest extends Specification {
```
Comment (peter-plochan) [inline gooddata-java/src/test/groovy/com/gooddata/sdk/service/export/ExportServiceTest.groovy]: There are no unit tests with valid arguments, e.g. for testing interaction with rest template. Create new github issue for this or add them in this PR.

Comment (jimirocks) [reply gooddata-java/src/test/groovy/com/gooddata/sdk/service/export/ExportServiceTest.groovy]: As there was this test before I only migrated it to Spock

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/110/ : SUCCESS in 32s

**Pull request merged**